### PR TITLE
Update wordpress/data-controls dependency to 2.6.1

### LINKED ISSUE #6075: Update `@wordpress/data-controls` dependency to version 2.6.1
### Description

We want to update the [@wordpress/data-controls](https://www.npmjs.com/package/@wordpress/data-controls) package to a more recent version, the one being used by our minimum supported version of WordPress:

- Current version: [1.6.0](https://www.npmjs.com/package/@wordpress/data-controls/v/1.6.0)
- Target version: [2.6.1](https://www.npmjs.com/package/@wordpress/data-controls/v/2.6.1)

### Acceptance criteria

- JS unit tests are passing
- For TypeScript libraries, ensure we're using the built-in types shipped with the target version instead of our custom types, if applicable.
  - If it's too much or it seems too complicated because of a lack of expertise in the relevant frontend areas, feel free to document what's missing to be confident about the update and ask for help from [the team owning this area](https://wootransactp2.wordpress.com/welcome-to-transact/product-responsibilities/)
- The review process from the [dependencies doc](https://github.com/Automattic/woocommerce-payments/blob/develop/docs/dependencies.md#javascript-dev-dependencies) was followed and now has an entry for this package with all the relevant details for future iterations

### Dev notes

Run `npm install --save-exact <packageName@version>` to update a package by an exact version.

Manual UI checks or/and E2E tests don't have any added value here because we're using the bundled version in WordPress, and updating the version locally will not impact the version used when using WCPay.

### PR DESCRIPTION
**Pull request by fashuvo**
Fixes #6075

#### Changes proposed in this Pull Request
- Update the version of `@wordpress/data-controls` to v2.6.1
- Add an entry in the dependencies doc

#### Testing instructions

* JS tests are passing

-------------------

- [x] Run `npm run changelog` to add a changelog file, choose `patch` to leave it empty if the change is not significant. You can add multiple changelog files in one PR by running this command a few times. 
- [x] Covered with tests (or have a good reason not to test in description ☝️)
- [ ] Tested on mobile (or does not apply)

**Post merge**

- [ ] Link to testing instructions from [release testing doc](https://github.com/Automattic/woocommerce-payments/wiki/Release-testing-instructions) following [these instructions](https://github.com/Automattic/woocommerce-payments/wiki/How-to-write-good-manual-testing-scenarios) : _Add link here / 'QA Testing Not Applicable'_
- [ ] Add or update [critical flows](https://github.com/Automattic/woocommerce-payments/wiki/Critical-flows) and [testing instructions for critical flows](https://github.com/Automattic/woocommerce-payments/wiki/Testing-instructions-for-critical-flows), if applicable.
- [ ] Add what's changed (description, screenshot, demo videos etc.) to the release announcement post, if applicable.

### CODE REVIEW

**Comment by botwoo** [comment]
### Test the build 

#### Option 1. Jetpack Beta

- Install and activate [Jetpack Beta](https://jetpack.com/download-jetpack-beta/).
- Use this build by searching for PR number `6256` or branch name `update/wordpress-data-controls-dependency` in your-test.site/wp-admin/admin.php?page=jetpack-beta&plugin=woocommerce-payments          

#### Option 2. Jurassic Ninja - available for logged-in A12s

:rocket: [Launch a JN site with this branch](https://jurassic.ninja/create/?jetpack-beta&shortlived&nojetpack&woocommerce&branches.woocommerce-payments=update/wordpress-data-controls-dependency) :rocket:

:information_source: Install this [Tampermonkey script](https://github.com/Automattic/woocommerce-payments/tree/develop/bin/wcpay-live-branches) to get more options.

---

Build info: 

- Latest commit: e3bbc7a5baed2daafe0cfa1856b2b3f599fbb5fc
- Build time: 2023-05-12 08:52:29 UTC

_Note: the build is updated when a new commit is pushed to this PR._

File: client/types/data.d.ts [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+	type ControlDescriptor = {
+		type: string;
+		storeKey: string | StoreDescriptor< AnyConfig >;
+		actionName: string;
+		args: unknown[];
+	};
+
+	interface Controls {
+		dispatch: (
+			storeKey: string,
```
Comment (jbordonado) [inline client/types/data.d.ts:17]: I think it's the other way around

For the input, it's typed `string | StoreDescriptor< AnyConfig >` and for the output, it's a `string`

Comment (jbordonado) [inline client/types/data.d.ts]: I think it's the other way around

For the input, it's typed `string | StoreDescriptor< AnyConfig >` and for the output, it's a `string`

**Review by jbordonado — CHANGES_REQUESTED** [review]
I left a small comment on a type. Can you also resolve the conflicts please?

Did you check a few flows manually to increase our confidence? For instance, is accepting a dispute (related to the change in `client/data/disputes/actions.js`) still working like before?

Since we replaced one function with another, we must also check the UI and its behavior. I would feel more confident if we tested a few of them manually. E2E tests are passing so it's a good sign already

**Comment by fashuvo** [comment]
> Did you check a few flows manually to increase our confidence? For instance, is accepting a dispute (related to the change in `client/data/disputes/actions.js`) still working like before?
> 
> Since we replaced one function with another, we must also check the UI and its behavior. I would feel more confident if we tested a few of them manually.

Yeah, I made some new transactions and a dispute. I was able to accept the dispute, see all the latest transactions, and see the list of all deposits as well.

File: client/types/data.d.ts (hunk 1 above)
Comment (jbordonado) [reply client/types/data.d.ts]: Is it resolved? Seems like the typing is the same as before

Comment (fashuvo) [reply client/types/data.d.ts]: Sorry, it took me a while to understand the input and output here. Resolved in [9a04a36](https://github.com/Automattic/woocommerce-payments/pull/6256/commits/9a04a36a6fe7759b2ab06a6386615c301e70f128)

Comment (jbordonado) [reply client/types/data.d.ts]: NP, thanks.

What I did to figure out the types is just dig through the package inside the `node_modules`, for reference
But I got it mixed up the first time on this field

**Review by jbordonado — APPROVED** [review]
I don't see any usage of the `dispatch` method from `@wordpress/data-controls` anymore, and the JS unit tests are passing.

LGTM

**Pull request merged**